### PR TITLE
feat: add wishlist component and functionality

### PR DESCRIPTION
**Pull request by Aditi0505**
# Change Logs:

# Description

- Added Wishlist Component
- Added Wishlist  Functionalities

    -    Add to Wishlist
    -    Move to Wishlist
    -    Remove from Wishlist
    -    Move to cart from Wishlist
    -    Increase Cart Total Amount on moving product from Wishlist  to Cart 
    -    Decrease Product Quantity on moving product from Cart to Wishlist 
  - Added Toast component to Wishlist
  - Fixed Toast functionality on Cart 
  - Fixed Cart total quantity if a product is already present

# Screenshots
image
image

# Preview Link

https://deploy-preview-5--redcloset.netlify.app/

### CODE REVIEW

File: src/components/Card/VerticalCard.jsx [hunk 1]
```diff
@@ -50,9 +50,29 @@ const VerticalCard = ({ product }) => {
         <button className="btn-action-horizontal-hide secondary">
           Remove from wishlist
         </button>
-        <span className="card-badge flex-center">
-          <i className="far fa-heart"></i>
-        </span>
+        <button className="card-badge flex-center">
+          {cartState.wishlistItems.some((i) => i._id === product._id) ? (
```
Comment (anupkgurung) [inline src/components/Card/VerticalCard.jsx:54]: instead of i you could use index or idx.

**Review by anupkgurung — APPROVED** [review]
[optional] - you could use increment and decrement operator instead of +1 /-1.
Rest code is clean and seems good.

File: src/pages/Wishlist/Wishlist.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useEffect } from "react";
+import { VerticalCard } from "../../components";
+import { useCart, useToast } from "../../context";
+import { setTitle } from "../../utils/set-title";
+const Wishlist = () => {
+  const title = "The Red Closet | Wishlist";
+  setTitle(title);
+  const { cartState } = useCart();
+  const { dispatch } = useToast();
+  console.log("wishlist", cartState.wishlistItems);
```
Comment (ranitasahaa1230) [inline src/pages/Wishlist/Wishlist.jsx:10]: Remove this console.log

**Review by ranitasahaa1230 — APPROVED** [review]
Good folder structure, code can be understood easily
 Looks good to merge

File: src/components/Card/VerticalCard.jsx (hunk 1 above)
Comment (Aditi0505) [reply src/components/Card/VerticalCard.jsx:54]: Sure, I will make the necessary change.

File: src/pages/Wishlist/Wishlist.jsx (hunk 2 above)
Comment (Aditi0505) [reply src/pages/Wishlist/Wishlist.jsx:10]: Sure, I will remove it.

**Pull request merged**